### PR TITLE
M07 hibernate

### PR DESCRIPTION
**Pull request by alextimonov**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -25,6 +26,8 @@
         <c3p0Version> 0.9.5.2 </c3p0Version>
         <springVersion> 4.3.1.RELEASE </springVersion>
         <javaVersion> 1.8 </javaVersion>
+        <hiberVersion> 5.2.1.Final </hiberVersion>
```
Comment (schaffe) [inline pom.xml]: what does it mean c3p0, hiber? Abbreviations are evil

File: src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/hibernate/HCookedDishDao.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import org.springframework.transaction.annotation.Transactional;
+import ua.goit.timonov.enterprise.module_6_2.dao.CookedDishDAO;
+import ua.goit.timonov.enterprise.module_6_2.model.*;
+
+import java.util.List;
+
+/**
+ * Hibernate implementation of CookedDishDAO
+ */
+public class HCookedDishDao implements CookedDishDAO {
```
Comment (schaffe) [inline src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/hibernate/HCookedDishDao.java]: Don't be stingy in letters :) 
Verbosity is better than incomprehension. Use full names

File: src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/hibernate/HCookedDishDao.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            int weightForDish = ingredientInDish.getIngredientWeight();
+            hStorageDao.changeAmount(ingredientInDish.getIngredient(), -weightForDish);
+        }
+    }
+
+    // checks if it's possible to add given dish to list of cooked dishes
+    private void checkAbilityToAdd(CookedDish cookedDish) {
+        Order order = cookedDish.getOrder();
+        Dish dish = cookedDish.getDish();
+        if (!order.getClosed()) {
```
Comment (schaffe) [inline src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/hibernate/HCookedDishDao.java]: a lot of curl braces pollutes code
be like Python ;)

File: src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/hibernate/HOrderDao.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    /**
+     * sets order to closed
+     * @param orderId
+     */
+    @Override
+    @Transactional(propagation = Propagation.MANDATORY)
+    public void setClosed(int orderId) {
+        if (orderIsClosed(orderId)) {
```
Comment (schaffe) [inline src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/hibernate/HOrderDao.java]: the same about curl braces

File: src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/jdbc/JdbcCookedDishDAO.java [hunk 5]
```diff
@@ -123,7 +123,7 @@ private Order defineOrderById(int orderId) {
         Map<String, Object> map = template.queryForMap(sql, orderId);
```
Comment (schaffe) [inline src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/jdbc/JdbcCookedDishDAO.java]: why map is `<String, Object>`?

File: pom.xml (hunk 1 above)
Comment (alextimonov) [reply pom.xml]: Fixed. c3p0 is an artifactId of a JDBC Connection pooling / Statement caching library com.mchange/c3p0. Changed to "comMchangeC3p0Version".

File: src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/hibernate/HCookedDishDao.java (hunk 2 above)
Comment (alextimonov) [reply src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/hibernate/HCookedDishDao.java]: Fixed

File: src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/hibernate/HCookedDishDao.java (hunk 3 above)
Comment (alextimonov) [reply src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/hibernate/HCookedDishDao.java]: Fixed. Deleted curl braces if there is only one statement in if-operator.

File: src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/hibernate/HOrderDao.java (hunk 4 above)
Comment (alextimonov) [reply src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/hibernate/HOrderDao.java]: Fixed

**Comment by schaffe** [comment]
Can be marged

**Pull request merged**